src/lxml.c

```diff
@@ -17,8 +18,85 @@
 #include <libxml/parser.h>
 #include <libxml/dict.h>
 
+#define XMLSEC_EXTRACT_VERSION(x, y) ((x / (y)) % 100)
+
+#define XMLSEC_EXTRACT_MAJOR(x) XMLSEC_EXTRACT_VERSION(x, 100 * 100)
+#define XMLSEC_EXTRACT_MINOR(x) XMLSEC_EXTRACT_VERSION(x, 100)
+#define XMLSEC_EXTRACT_PATCH(x) XMLSEC_EXTRACT_VERSION(x, 1)
+
+static long PyXmlSec_GetLibXmlVersionLong() {
+    return PyOS_strtol(xmlParserVersion, NULL, 10);
+}
+long PyXmlSec_GetLibXmlVersionMajor() {
+    return XMLSEC_EXTRACT_MAJOR(PyXmlSec_GetLibXmlVersionLong());
+}
+long PyXmlSec_GetLibXmlVersionMinor() {
+    return XMLSEC_EXTRACT_MINOR(PyXmlSec_GetLibXmlVersionLong());
+}
+long PyXmlSec_GetLibXmlVersionPatch() {
+    return XMLSEC_EXTRACT_PATCH(PyXmlSec_GetLibXmlVersionLong());
+}
+
+long PyXmlSec_GetLibXmlCompiledVersionMajor() {
+    return XMLSEC_EXTRACT_MAJOR(LIBXML_VERSION);
+}
+long PyXmlSec_GetLibXmlCompiledVersionMinor() {
+    return XMLSEC_EXTRACT_MINOR(LIBXML_VERSION);
+}
+long PyXmlSec_GetLibXmlCompiledVersionPatch() {
+    return XMLSEC_EXTRACT_PATCH(LIBXML_VERSION);
+}
+
+static int PyXmlSec_CheckLxmlLibraryVersion(void) {
+    // Make sure that the version of libxml2 lxml is using is the same as the one we are using. Because
+    // we pass trees between the two libraries, we need to make sure that they are using the same version
+    // of libxml2, or we could run into difficult to debug segfaults.
+    // See: https://github.com/xmlsec/python-xmlsec/issues/283
+
+    PyObject* lxml = NULL;
+    PyObject* version = NULL;
+
+    // Default to failure
+    int result = -1;
+
+    lxml = PyImport_ImportModule("lxml.etree");
+    if (lxml == NULL) {
+        goto FINALIZE;
+    }
+    version = PyObject_GetAttrString(lxml, "LIBXML_VERSION");
+    if (version == NULL) {
+        goto FINALIZE;
+    }
+    if (!PyTuple_Check(version) || PyTuple_Size(version) != 3) {
```

Comment:
It's probably enough to check "size < 2" here, if all you look at is the first two entries.